pkg/datapath/loader/base.go

```diff
@@ -0,0 +1,303 @@
+		} else {
+			args[initArgHostReachableServicesUDP] = "false"
+		}
+	} else {
+		args[initArgHostReachableServices] = "false"
+		args[initArgHostReachableServicesUDP] = "false"
+	}
+
+	if option.Config.EncryptInterface != "" {
+		args[initArgEncryptInterface] = option.Config.EncryptInterface
+	}
+
+	if option.Config.Device != "undefined" {
+		_, err := netlink.LinkByName(option.Config.Device)
+		if err != nil {
+			log.WithError(err).WithField("device", option.Config.Device).Warn("Link does not exist")
+			return err
+		}
+
+		if option.Config.DatapathMode == option.DatapathModeIpvlan {
+			mode = "ipvlan"
+		} else {
+			mode = "direct"
+		}
+
+		args[initArgMode] = mode
+		if option.Config.EnableNodePort &&
+			strings.ToLower(option.Config.Tunnel) != "disabled" {
+			args[initArgMode] = option.Config.Tunnel
+		}
+		args[initArgDevice] = option.Config.Device
+	} else {
+		args[initArgMode] = option.Config.Tunnel
+
+		if option.Config.IsFlannelMasterDeviceSet() {
+			args[initArgMode] = "flannel"
+			args[initArgDevice] = option.Config.FlannelMasterDevice
+		}
+	}
+
+	if option.Config.EnableEndpointRoutes == true {
+		args[initArgMode] = "routed"
+	}
+
+	if option.Config.EnableNodePort {
+		args[initArgNodePort] = "true"
+	}
+
+	log.Info("Setting up base BPF datapath")
+
+	for _, s := range sysSettings {
+		log.Infof("Setting sysctl %s=%s", s.name, s.val)
+		if err := sysctl.Write(s.name, s.val); err != nil {
+			if !s.ignoreErr {
+				return fmt.Errorf("Failed to sysctl -w %s=%s: %s", s.name, s.val, err)
+			}
+			log.WithError(err).WithFields(logrus.Fields{
+				logfields.SysParamName:  s.name,
+				logfields.SysParamValue: s.val,
+			}).Warning("Failed to sysctl -w")
+		}
+	}
+
+	prog := filepath.Join(option.Config.BpfDir, "init.sh")
+	ctx, cancel := context.WithTimeout(context.Background(), defaults.ExecTimeout)
+	defer cancel()
+	cmd := exec.CommandContext(ctx, prog, args...)
+	cmd.Env = bpf.Environment()
+	if _, err := cmd.CombinedOutput(log, true); err != nil {
+		return err
+	}
+
+	if l.canDisableDwarfRelocations {
+		// Validate alignments of C and Go equivalent structs
+		if err := alignchecker.CheckStructAlignments(defaults.AlignCheckerName); err != nil {
+			log.WithError(err).Fatal("C and Go structs alignment check failed")
+		}
+	} else {
+		log.Warning("Cannot check matching of C and Go common struct alignments due to old LLVM/clang version")
+	}
+
+	if !option.Config.IsFlannelMasterDeviceSet() {
+		r.ReserveLocalRoutes()
+	}
+
+	if err := o.Datapath().Node().NodeConfigurationChanged(*o.LocalConfig()); err != nil {
+		return err
+	}
+
+	if option.Config.InstallIptRules {
+		if err := iptMgr.TransientRulesStart(option.Config.HostDevice); err != nil {
+			return err
+		}
+	}
+	// Always remove masquerade rule and then re-add it if required
+	iptMgr.RemoveRules()
+	if option.Config.InstallIptRules {
+		err := iptMgr.InstallRules(option.Config.HostDevice)
+		iptMgr.TransientRulesEnd(false)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
The above bits feel like they should be able to live entirely in the iptables rule manager.

Reply:
(as it turns out, the proxy bits are also about managing iptables but I guess you'd have to come up with a wrapper type somewhere if we wanted to combine these two in the inputs to this function)

Reply:
Agreed - I think the proxy bits should be taken care of entirely by some entity which manages iptables (which is another refactor).